File: Java/exercises/Quadratic/Quadratic.pl
# Quadratic.pl

use v6-alpha;

@*ARGS == 3 or
    die "Usage: $?FILE <a> <b> <c>\n";
my ($a, $b, $c) = @*ARGS;

if $a == 0 { die "Error: It's not a quadratic equation.\n" }

my $delta = $b * $b - 4 * $a * $c;
if $delta < 0 { die "Error: No solution found.\n" }

my $p = -$b / (2*$a);
my $q = sqrt($delta) / (2*$a);

my ($x1, $x2) = $p + $q, $p - $q;

say "Solution found: $x1, $x2";

File: Java/exercises/Quadratic/Quadratic.java
// Quadratic.java

public class Quadratic {
    public static void main(String[] args) {
        if (args.length != 3)
            die("Usage: java Quadratic <a> <b> <c>");
        double a = Double.parseDouble(args[0]);
        double b = Double.parseDouble(args[1]);
        double c = Double.parseDouble(args[2]);
        if (a == 0)
            die("Error: It's not a valid quadratic equation.");

        double delta = b*b - 4*a*c;
        if (delta < 0)
            die("Error: No solution found.");

        double p = -b / (2*a);
        double q = Math.sqrt(delta) / (2*a);
        
        double x1 = p + q;
        double x2 = p - q;
        System.out.println("Solution Found: " + x1 + ", " + x2);
    }
    private static void die(String s) {
        System.err.println(s);
        System.exit(1);
    }
}
